src/Parser/Parser.hs

```diff
@@ -176,18 +188,13 @@ notAlreadyInUse x = do
     then unexpected $ "redefinition of " ++ parsed
     else return parsed
 
--- | Valid character set that can be used to describe a game (after the first lead character)
-gameIdentifierChars :: [Char]
-gameIdentifierChars = ['a'..'z']++['A'..'Z']++['0'..'9']++"_"
-
--- | Identifies a valid game name, also used for types
--- nearly identical to 'capIdentifier', just includes underscores
+-- | Identifies a valid game name
 gameIdentifier :: ParsecT String u Identity [Char]
-gameIdentifier = lexeme ((:) <$> upper <*> (many (oneOf gameIdentifierChars)))
+gameIdentifier = capIdentifier
 
 -- | Starting uppercase letter identifier
 capIdentifier :: ParsecT String u Identity [Char]
-capIdentifier = gameIdentifier
+capIdentifier = lookAhead upper *> identifier
```

Comment:
this allowed, for example, `type T = {True, Board, Int}`

Reply:
Ah, so it wasn't screening the names, yeah that's a good catch.